### PR TITLE
CLOUDSTACK-10012: Jetty 9.4

### PR DESCRIPTION
**Pull request by marcaurele**
Upgrading to jetty 9.4 along with:
- new gzip handler
- redirect `/` to the context path
- add session timeout configuration
- change for cloudian tests

### CODE REVIEW

**Comment by marcaurele** [comment]
superseed #2328 (Jetty 9.4 is Java8 compatible)
cc @rhtyd @wido

File: client/conf/server.properties.in [hunk 1]
```diff
@@ -24,6 +24,9 @@ context.path=/client
 # The HTTP port to be used by the management server
 http.port=8080
 
+# Max inactivity time in minutes for the session
+session.timeout=10
```
Comment (rhtyd) [inline client/conf/server.properties.in]: @marcaurele Can we bump the default to 30 minutes? I suppose 10 minutes might be too low, or maybe we can hit the sweet spot at 20mins?

Comment (wido) [reply client/conf/server.properties.in]: I agree that 10 minutes is very short. I would prefer 30

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@marcaurele I've fixed the cloudian unit test failures and the session timeout to 30 (mins).

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@marcaurele how do you start the jetty server with mvn, when I run `mvn -pl :cloud-client-ui jetty:run  -Dnoredist -Dsimulator -Djava.net.preferIPv4Stack=true` it does not run the server for me.

**Comment by rhtyd** [comment]
@marcaurele I found a workaround to use the old 9.2.x jetty-maven-plugin as the newer one did not work for me (and Travis). Please review the changes.
LGTM from my side.

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by marcaurele** [comment]
I moved the plugin version to root pom `properties`.

**Comment by marcaurele** [comment]
LGTM

File: client/src/org/apache/cloudstack/ServerDaemon.java [hunk 2]
```diff
@@ -129,6 +126,7 @@ public void init(final DaemonContext context) {
             setKeystorePassword(properties.getProperty(KEYSTORE_PASSWORD));
             setWebAppLocation(properties.getProperty(WEBAPP_DIR));
             setAccessLogFile(properties.getProperty(ACCESS_LOG, "access.log"));
+            setSessionTimeout(Integer.valueOf(properties.getProperty(SESSION_TIMEOUT, "10")));
```
Comment (rafaelweingartner) [inline client/src/org/apache/cloudstack/ServerDaemon.java]: @marcaurele you have a default value in the property file, which is 30, and here it is 10 (at line 85). Moreover, what about using the value of variable `sessionTimeout ` as the default values? The variable already starts with the default value anyways.

Comment (rhtyd) [reply client/src/org/apache/cloudstack/ServerDaemon.java]: Thanks, I've synced default values to 30 now.

**Comment by rhtyd** [comment]
@rafaelweingartner I'll squash merge them (I dislike splitting PRs in 10s+ of commits too)

**Comment by rafaelweingartner** [comment]
@rhtyd thanks!

**Comment by rhtyd** [comment]
Tests LGTM, merged. Thanks everyone.

**Comment by andrijapanicsb** [comment]
Resurrecting an old PR - did anyone actually tested the session duration being set correctly (to something else than 30min)?

It's not working in 4.11+ (whatever other value you set, the session is always set to 30min)
Tested by setting session.timeout=2 in the server.properties, restarting mgmt server.

@marcaurele @wido @rafaelweingartner  /CC @rhtyd

**Comment by marcaurele** [comment]
@andrijapanicsb I'll have a look and come back here with the fix

**Pull request merged**